builtin/logical/kv/path_metadata_test.go

```diff
@@ -1043,6 +1162,63 @@ func TestVersionedKV_Metadata_Patch_CasRequiredWarning(t *testing.T) {
 	}
 }
 
+func TestVersionedKV_Metadata_Patch_MetadataCasRequiredWarning(t *testing.T) {
+	b, storage := getBackend(t)
+
+	// Set global metadata_cas_required
+	req := &logical.Request{
+		Operation: logical.CreateOperation,
+		Path:      "config",
+		Storage:   storage,
+		Data: map[string]interface{}{
+			"metadata_cas_required": true,
+		},
+	}
+
+	resp, err := b.HandleRequest(context.Background(), req)
+	if err != nil || (resp != nil && resp.IsError()) {
+		t.Fatalf("config request failed, err:%s resp:%#v\n", err, resp)
+	}
+
+	// Create metadata
+	req = &logical.Request{
+		Operation: logical.CreateOperation,
+		Path:      "metadata/foo",
+		Storage:   storage,
+		Data: map[string]interface{}{
+			"max_versions": 5,
+			"metadata_cas": 0, // Required because global setting
+		},
+	}
+
+	resp, err = b.HandleRequest(context.Background(), req)
+	if err != nil || (resp != nil && resp.IsError()) {
+		t.Fatalf("metadata create request failed, err:%s resp:%#v\n", err, resp)
+	}
+
+	// Try to disable metadata_cas_required via PATCH
+	req = &logical.Request{
+		Operation: logical.PatchOperation,
+		Path:      "metadata/foo",
+		Storage:   storage,
+		Data: map[string]interface{}{
+			"metadata_cas_required": false,
+			"metadata_cas":          0,
```

Comment:
```suggestion
			"metadata_cas":          1,
```

Shouldn't this be 2 though? :thinking: We should've set 1 earlier, so I would've expected this to fail since we updated it above?